src/treetensornetworks/solvers/tdvp.jl

```diff
@@ -30,56 +30,78 @@ function sub_time_steps(order)
-  reverse_step=true,
-)
-  sweep_plan = []
-  for (substep, fac) in enumerate(sub_time_steps(order))
-    sub_time_step = time_step * fac
-    half = half_sweep(
-      direction(substep),
-      graph,
-      make_region;
-      root_vertex,
-      nsites,
-      region_args=(; substep, time_step=sub_time_step),
-      reverse_args=(; substep, time_step=-sub_time_step),
-      reverse_step,
-    )
-    append!(sweep_plan, half)
-  end
-  return sweep_plan
-end
+"""
+    tdvp(operator::TTN, t::Number, init_state::TTN; kwargs...)
 
+Use the time dependent variational principle (TDVP) algorithm
+to approximately compute `exp(operator*t)*init_state` using an efficient algorithm based
+on alternating optimization of the state tensors and local Krylov
+exponentiation of operator. The time parameter `t` can be a real or complex number.
+                    
+Returns:
+* `state` - time-evolved state
+
+Optional keyword arguments:
+* `time_step::Number = t` - time step to use when evolving the state. Smaller time steps generally give more accurate results but can make the algorithm take more computational time to run.
+* `nsteps::Integer` - evolve by the requested total time `t` by performing `nsteps` of the TDVP algorithm. More steps can result in more accurate results but require more computational time to run. (Note that only one of the `time_step` or `nsteps` parameters can be provided, not both.)
+* `outputlevel::Int = 1` - larger outputlevel values resulting in printing more information and 0 means no output
+* `observer` - object implementing the Observer interface which can perform measurements and stop early
+* `write_when_maxdim_exceeds::Int` - when the allowed maxdim exceeds this value, begin saving tensors to disk to free memory in large calculations
+"""
 function tdvp(
-  updater,
   operator,
   t::Number,
   init_state::AbstractTTN;
   time_step::Number=t,
   nsites=2,
   nsteps=nothing,
   order::Integer=2,
+  outputlevel=0,
   (sweep_observer!)=observer(),
   root_vertex=default_root_vertex(init_state),
   reverse_step=true,
+  extracter_kwargs=(;),
+  extracter=default_extractor(), # ToDo: extracter could be inside extracter_kwargs, at the cost of having to extract it in region_update
   updater_kwargs=(;),
+  updater=exponentiate_updater,
+  inserter_kwargs=(;),
+  inserter=default_inserter(),
   kwargs...,
 )
+  # slurp unbound kwargs into inserter
+  ### if unbound kwargs are required in e.g. updater, they will have to be explicitly listed inside e.g. updater_kwargs
+  ### if non-standard inserter (with different kwarg signature) is used, it is up to the user to not pass unsupported kwargs
+  inserter_kwargs = (; inserter_kwargs..., kwargs...) # slurp unbound kwargs into inserter
+  # move inserter etc. into the respective kwargs
+  inserter_kwargs = merge((; inserter), inserter_kwargs)
+  updater_kwargs = merge((; updater), updater_kwargs)
+  extracter_kwargs = merge((; extracter), extracter_kwargs)
```

Comment:
Don't merge the functions into the named tuples in this way, instead just pass to `process_kwargs_for_sweeps` (or whatever that will be called).